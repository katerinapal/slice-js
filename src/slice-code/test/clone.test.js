import {runAllCombosTests} from './helpers/utils'

// this is here to make it easy to isolate tests to a specific case
/*
test.only(
  'testing uncoveredConditionalExpression(false)',
  require('./helpers/utils').snapSlice(
    require.resolve('./fixtures/cond-expr'),
    ({uncoveredConditionalExpression}) => {
      uncoveredConditionalExpression(false)
    },
  ),
)
/* */

runAllCombosTests({
  filename: require.resolve('./fixtures/clone'),
  methods: [
    {
      useDefaultExport: true,
      methodName: 'clone',
      possibleArguments: [
        ['hello'],
        [null],
        [{name: 'Luke'}],
        [{friends: [{name: 'Rebecca'}]}],
        [
          {
            title: 'How to Train Your Dragon',
            releaseDate: new Date('2010-03-26 00:00:00'),
          },
        ],
      ],
    },
  ],
})
