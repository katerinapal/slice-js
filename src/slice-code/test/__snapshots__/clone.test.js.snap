exports[`test clone("hello") && clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone({"name":"Luke"}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone({"name":"Luke"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    result = {};
    for (const i in item) {
      result[i] = clone(item[i]);
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) && clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) && clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) && clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) && clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) && clone({"name":"Luke"}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) && clone({"name":"Luke"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    result = {};
    for (const i in item) {
      result[i] = clone(item[i]);
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone("hello") && clone(null) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }

  Object.prototype.toString.call(item);

  return item;
}"
`;

exports[`test clone("hello") 1`] = `
"export default clone;

function clone(item) {
  Object.prototype.toString.call(item);

  return item;
}"
`;

exports[`test clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone({"name":"Luke"}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone({"name":"Luke"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    result = {};
    for (const i in item) {
      result[i] = clone(item[i]);
    }
  }

  return result;
}"
`;

exports[`test clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone(null) && clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone(null) && clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone(null) && clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      if (string === \'[object Date]\') {
        result = new Date(item);
      } else {
        result = {};
        for (const i in item) {
          result[i] = clone(item[i]);
        }
      }
    }
  }

  return result;
}"
`;

exports[`test clone(null) && clone({"name":"Luke"}) && clone({"friends":[{"name":"Rebecca"}]}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Array]\') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone(null) && clone({"name":"Luke"}) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone(null) && clone({"name":"Luke"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    result = {};
    for (const i in item) {
      result[i] = clone(item[i]);
    }
  }

  return result;
}"
`;

exports[`test clone(null) && clone({"title":"How to Train Your Dragon","releaseDate":"2010-03-26T07:00:00.000Z"}) 1`] = `
"export default clone;

function clone(item) {
  if (!item) {
    return item;
  }
  const type = typeof item;
  const string = Object.prototype.toString.call(item);

  let result = item;

  if (!(type !== \"object\" && type !== \"function\")) {
    if (string === \'[object Date]\') {
      result = new Date(item);
    } else {
      result = {};
      for (const i in item) {
        result[i] = clone(item[i]);
      }
    }
  }

  return result;
}"
`;

exports[`test clone(null) 1`] = `
"export default clone;

function clone(item) {
  return item;
}"
`;
